shims/spark311/src/main/scala/com/nvidia/spark/rapids/shims/spark311/ParquetCachedBatchSerializer.scala

```diff
@@ -849,13 +849,12 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
               throw new UnsupportedOperationException("Schema evolution not supported.")
             }
             missingColumns(i) = false
-          }
-          else {
+          } else {
             if (columns.get(i).getMaxDefinitionLevel == 0) {
               // Column is missing in data but the required data is non-nullable.
               // This file is invalid.
-              throw new IOException("Required column is missing in data file. Col: "
-                + colPath.toList)
+              throw new IOException(s"Required column is missing in data file. Col: " +
+                s"${colPath.toList}")
```

Comment:
There's no point in formatting a string that doesn't have format directives, and the `Col: ` is redundant since the message already mentions a column is missing.
```suggestion
              throw new IOException(s"Required column is missing in data file: ${colPath.toList}")
```